homeassistant/components/screenlogic/services.py

```diff
@@ -61,10 +84,36 @@ async def extract_screenlogic_config_entry_ids(service_call: ServiceCall):
             )
         return screenlogic_entry_ids
 
+    async def get_coordinators(
+        service_call: ServiceCall,
+    ) -> list[ScreenlogicDataUpdateCoordinator]:
+        if entry_id := service_call.data.get(ATTR_CONFIG_ENTRY):
+            entry_ids = [entry_id]
+        else:
+            ir.async_create_issue(
+                hass,
+                DOMAIN,
+                "service_target_deprecation",
+                breaks_in_ha_version="2024.7.0",
+                is_fixable=True,
+                is_persistent=True,
+                severity=ir.IssueSeverity.WARNING,
+                translation_key="service_target_deprecation",
+            )
+            entry_ids = await extract_screenlogic_config_entry_ids(service_call)
+
+        if not (
+            coordinators := [hass.data[DOMAIN].get(entry_id) for entry_id in entry_ids]
```

Comment:
I think we should check the state of the config entry when extracting the entry ids so we can use `dict[key]` here. That's more robust. The assumption is that if the config entry is loaded, there should be a coordinator in hass.data.